### PR TITLE
Headless Chrome: keep browser alive between requests.

### PR DESCRIPTION
**Pull request by steren**
Start the browser and page once, and reuse them for each request.
This improves the app's performance

### CODE REVIEW

**Comment by steren** [comment]
(Ignore the many commits, it's just that I did not rebase from head and had to merge my previous branch)

File: appengine/headless-chrome/app.js [hunk 1]
```diff
@@ -20,32 +20,36 @@ const express = require('express');
...
   res.set('Content-Type', 'image/png');
   res.send(imageBuffer);
 });
 
-const server = app.listen(process.env.PORT || 8080, err => {
-  if (err) return console.error(err);
-  const port = server.address().port;
-  console.info(`App listening on port ${port}`);
-});
+async function init () {
```
Comment (fhinkel) [inline appengine/headless-chrome/app.js]: Can we move this further up so it's above its first use? It's hard to read this way.

File: appengine/headless-chrome/app.js [hunk 2]
```diff
@@ -20,32 +20,36 @@ const express = require('express');
...
+  // [START browser]
+  const browser = await puppeteer.launch({
+    args: ['--no-sandbox']
+  });
+  // [END browser]
+  page = await browser.newPage();
+  const server = app.listen(process.env.PORT || 8080, err => {
+    if (err) return console.error(err);
```
Comment (fhinkel) [inline appengine/headless-chrome/app.js]: Shouldn't we call `browser.close()` on error?

File: appengine/headless-chrome/app.js (hunk 1 above)
Comment (steren) [reply appengine/headless-chrome/app.js]: Done

File: appengine/headless-chrome/app.js (hunk 2 above)
Comment (steren) [reply appengine/headless-chrome/app.js]: Done

File: appengine/headless-chrome/app.js [hunk 3]
```diff
@@ -20,32 +20,40 @@ const express = require('express');
...
+async function init () {
+  // [START browser]
+  const browser = await puppeteer.launch({
+    args: ['--no-sandbox']
+  }); 
+  // [END browser]
+  page = await browser.newPage();
+  const server = app.listen(process.env.PORT || 8080, err => {
+    if (err) {
+      browser.close();
```
Comment (ebidel) [inline appengine/headless-chrome/app.js]: `await browser.close()`

File: appengine/headless-chrome/app.js [hunk 4]
```diff
@@ -20,32 +20,40 @@ const express = require('express');
...
+let page;
+
+async function init () {
+  // [START browser]
+  const browser = await puppeteer.launch({
+    args: ['--no-sandbox']
+  }); 
+  // [END browser]
+  page = await browser.newPage();
+  const server = app.listen(process.env.PORT || 8080, err => {
```
Comment (ebidel) [inline appengine/headless-chrome/app.js]: `async error => {`

**Comment by steren** [comment]
Please do not merge yet: I realize that what should be shared is the browser object, not the page object.

**Comment by steren** [comment]
I fixed the issue. Please review and merge.

**Comment by fhinkel** [comment]
Is the failing linter a legitimate error?

**Pull request merged**